### PR TITLE
hotfix auth.service to provide necessary getters

### PR DESCRIPTION
**Pull request by mercuryedge117**

### CODE REVIEW

File: Chirp/src/app/shared/services/auth.service.ts
Comment (Jingyued) [inline Chirp/src/app/shared/services/auth.service.ts:1]: wondering why those are optional?
  private token?: string;
  private role?: string;
  private name?: string;

Comment (mercuryedge117) [reply Chirp/src/app/shared/services/auth.service.ts:1]: they are used for status management, if user weren't login yet, router guard could reference to these variable to block invalid access

Comment (Jingyued) [reply Chirp/src/app/shared/services/auth.service.ts:1]: what's the use case of user's role being undefined ——  I thought if user successfully logged in, his role should be either admin or user

Comment (mercuryedge117) [reply Chirp/src/app/shared/services/auth.service.ts:1]: > what's the use case of user's role being undefined —— I thought if user successfully logged in, his role should be either admin or user

If user tried to access admin page before they logged in (via directly entering url, which will skip our blocker on navbar), user variable will be accessed before it was initialized by login function. That's why I keep all of the gettable variable as optional, for both edge condition handling and status management

Comment (Jingyued) [reply Chirp/src/app/shared/services/auth.service.ts:1]: you're right. PR approved

**Review by Jingyued — APPROVED** [review]

**Pull request merged**